ng2-components/ng2-alfresco-core/src/dialogs/folder.dialog.ts

```diff
@@ -0,0 +1,141 @@
+    selector: 'adf-folder-dialog',
+    styleUrls: ['./folder.dialog.scss'],
+    templateUrl: './folder.dialog.html'
+})
+export class FolderDialogComponent implements OnInit {
+    form: FormGroup;
+    folder: MinimalNodeEntryEntity = null;
+
+    constructor(
+        private formBuilder: FormBuilder,
+        private dialog: MatDialogRef<FolderDialogComponent>,
+        private nodesApi: NodesApiService,
+        private translation: TranslationService,
+        private notification: NotificationService,
+        @Optional()
+        @Inject(MAT_DIALOG_DATA)
+        public data: any
+    ) {}
+
+    get editing(): boolean {
+        return !!this.data.folder;
+    }
+
+    ngOnInit() {
+        const { folder } = this.data;
+        let name = '', description = '';
+
+        if (folder) {
+            const { properties } = folder;
+
+            name = folder.name || '';
+            description = properties ? properties['cm:description'] : '';
+        }
+
+        const validators = {
+            name: [
+                Validators.required,
+                forbidSpecialCharacters,
+                forbidEndingDot,
+                forbidOnlySpaces
+            ]
+        };
+
+        this.form = this.formBuilder.group({
+            name: [ name, validators.name ],
+            description: [ description ]
+        });
+    }
+
+    get name(): string {
+        let { name } = this.form.value;
+
+        return (name || '').trim();
+    }
+
+    get description(): string {
+        let { description } = this.form.value;
+
+        return (description || '').trim();
+    }
+
+    private get properties(): any {
+        const { name: title, description } = this;
+
+        return {
+            'cm:title': title,
+            'cm:description': description
+        };
+    }
+
+    private create(): Observable<MinimalNodeEntryEntity> {
+        const { name, properties, nodesApi, data: { parentNodeId} } = this;
+        return nodesApi.createFolder(parentNodeId, { name, properties });
+    }
+
+    private edit(): Observable<MinimalNodeEntryEntity> {
+        const { name, properties, nodesApi, data: { folder: { id: nodeId }} } = this;
+        return nodesApi.updateNode(nodeId, { name, properties });
+    }
+
+    submit() {
+        const { form, dialog, editing } = this;
+
+        if (!form.valid) { return; }
+
+        (editing ? this.edit() : this.create())
+            .subscribe(
+                (folder: MinimalNodeEntryEntity) => dialog.close(folder),
+                (error) => this.handleError(error)
+            );
+    }
+
+    handleError(error: any): any {
+        let i18nMessageString = 'CORE.MESSAGES.ERRORS.GENERIC';
+
+        try {
+            const { error: { statusCode } } = JSON.parse(error.message);
+
+            if (statusCode === 409) {
+                i18nMessageString = 'CORE.MESSAGES.ERRORS.EXISTENT_FOLDER';
+            }
+        } catch (err) { /* Do nothing, keep the original message */ }
```

Comment:
please remove this comment

Reply:
I tried to simplify the code so there is no more try catch in there… but the JSON.parse will throw an error on any incorrect syntax encountered, so the try catch is needed.